### PR TITLE
Run Travis CI only on master, tag or release branch

### PR DESCRIPTION
**Pull request by dhiller**
**What this PR does / why we need it**:

As we have moved the checks to prow presubmits we do not want to run the
Travis builds on PRs against master any more.

Also remove sudo flag as it has no effect for quite a while now:
https://blog.travis-ci.com/2018-11-19-required-linux-infrastructure-migration

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

Should only get merged after https://github.com/kubevirt/project-infra/pull/651 is merged.

**Release note**:

```release-note
Only run Travis build for PRs against release branches
```

### CODE REVIEW

**Comment by dhiller** [comment]
Closing this as it's not required.

File: .travis.yml [hunk 1]
```diff
@@ -1,5 +1,6 @@
+if: type != pull_request OR branch =~ /^(release-[0-9\.]+)$/
+
 language: generic
-sudo: required
```
Comment (rmohr) [inline .travis.yml:2]: Why do we remove `sudo: required` here?

Comment (rmohr) [reply .travis.yml:2]: Never mind.

**Pull request merged**